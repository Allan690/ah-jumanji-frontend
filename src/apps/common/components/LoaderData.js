
import React, { Component } from 'react';

import 'masonry-layout';

class LoaderData extends Component {

  render() {
    return (
      <div className="grid" id="articles-data" data-masonry='{"itemSelector": ".grid-item", "columnWidth": ".grid-sizer"}'>
        <div className="grid-sizer" />
        <span id="loader-data" className="loader-data">
          <div className="grid-item">
            <div className="animated-background one">
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
            </div>
            <br />
            <div className="row">
              <div className="col-md-2">
                <div className="animated-background two">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
              <div className="col-md-10">
                <div className="animated-background three">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
            </div>
          </div>

          <div className="grid-item">
            <div className="animated-background four">
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
            </div>
            <br />
            <div className="row">
              <div className="col-md-2">
                <div className="animated-background five">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
              <div className="col-md-10">
                <div className="animated-background six">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
            </div>
          </div>

          <div className="grid-item">
            <div className="animated-background seven">
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
            </div>
            <br />
            <div className="row">
              <div className="col-md-2">
                <div className="animated-background eight">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
              <div className="col-md-10">
                <div className="animated-background nine">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
            </div>
          </div>

          <div className="grid-item">
            <div className="animated-background ten">
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
            </div>
            <br />
            <div className="row">
              <div className="col-md-2">
                <div className="animated-background eleven">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
              <div className="col-md-10">
                <div className="animated-background twelve">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
            </div>
          </div>

          <div className="grid-item">
            <div className="animated-background thirteen">
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
            </div>
            <br />
            <div className="row">
              <div className="col-md-2">
                <div className="animated-background fourteen">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
              <div className="col-md-10">
                <div className="animated-background fifteen">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
            </div>
          </div>


          <div className="grid-item">
            <div className="animated-background sixteen">
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
              <div className="background-masker header-top" />
              <div className="background-masker header-left" />
              <div className="background-masker header-right" />
              <div className="background-masker header-bottom" />
              <div className="background-masker subheader-left" />
              <div className="background-masker subheader-right" />
              <div className="background-masker subheader-bottom" />
              <div className="background-masker content-top" />
              <div className="background-masker content-first-end" />
              <div className="background-masker content-second-line" />
              <div className="background-masker content-second-end" />
              <div className="background-masker content-third-line" />
              <div className="background-masker content-third-end" />
            </div>
            <br />
            <div className="row">
              <div className="col-md-2">
                <div className="animated-background seventeen">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
              <div className="col-md-10">
                <div className="animated-background eighteen">
                  <div className="background-masker header-top" />
                  <div className="background-masker header-left" />
                  <div className="background-masker header-right" />
                  <div className="background-masker header-bottom" />
                  <div className="background-masker subheader-left" />
                  <div className="background-masker subheader-right" />
                  <div className="background-masker subheader-bottom" />
                  <div className="background-masker content-top" />
                  <div className="background-masker content-first-end" />
                  <div className="background-masker content-second-line" />
                  <div className="background-masker content-second-end" />
                  <div className="background-masker content-third-line" />
                  <div className="background-masker content-third-end" />
                </div>
              </div>
            </div>
          </div>
        </span>
      </div>
    );
  }
}

export default LoaderData;
