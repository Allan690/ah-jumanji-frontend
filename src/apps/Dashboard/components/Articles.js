import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import Masonry from 'masonry-layout';
import imagesLoaded from 'imagesloaded';
import { toast } from 'react-toastify';
import 'jquery';
import PropTypes from 'prop-types';
import { bake_cookie, read_cookie } from 'sfcookies';

import * as dashboardActions from '../actions/dashboardActions';

import LoaderData from '../../common/components/LoaderData';
import defaultUserIcon from '../../../assets/img/default-avatar.jpg';
import articlePlaceholder from '../../../assets/img/placeholder.png';
import ArticleComponent from './ArticleComponent';


class Articles extends Component {

  constructor(props) {
    super(props);
    this.state = {
      currentPage: this.props.currentPage,
      newPage: this.props.currentPage
    };
  }

  componentDidMount() {
    let onLogin = read_cookie('onLogin');
    if (onLogin === true) {
      toast.success( "Login Successful!", {autoClose: 3000});
    }

    bake_cookie('onLogin', false);

    const {actions} = this.props;
    const {currentPage} = this.state;
    actions.fetchArticle(`?page=${currentPage}`);
  }

  static getDerivedStateFromProps(nextProps, prevState){
    if (nextProps.currentPage!==prevState.currentPage){
      return {currentPage: nextProps.currentPage};
    }
    else return null;
  }

  componentDidUpdate(prevProps, prevState) {
    const {currentPage} = this.state;
    if (prevState.currentPage !== currentPage) {
      const {actions} = this.props;
      const {currentPage} = this.state;
      actions.fetchArticle(`?page=${currentPage}`);
    }
  }



  data = () => {
    const { articles } = this.props;

    if (articles !== undefined) {
      const f = articles.articles;
      const r = [];
      let counter = 0;
      let articleImg = '';

      for (const i in f) {
        if (f[i].body.indexOf('src') > -1) {
          const imageSplit = f[i].body.substring(f[i].body.indexOf('src=') + 1);
          articleImg = imageSplit.split(/"/)[1];
        } else {
          articleImg = articlePlaceholder;
        }

        const t = (
          <ArticleComponent
            key={i}
            articleImage={articleImg}
            articleSlug={f[i].slug}
            articleLikes={f[i].likes}
            articleComments={f[i].comments}
            articleRatings={f[i].ratings}
            articleAvatar={f[i].author.profile_photo !== '' ? f[i].author.profile_photo : defaultUserIcon}
            authorName={f[i].author.user}
            articleTitle={f[i].title}
            bookmarked={f[i].bookmarked}
          />
        );
        r.push(t);
        counter += 1;
      }
      if (counter==12) {
        return (r);
      }

    }
  };

  masonryLoader = () => {
    return (imagesLoaded('.grid', function() {
      let msnry = new Masonry( '.grid',{
        itemSelector: '.grid-item',
        columnWidth: '.grid-sizer'
      });
      msnry.reloadItems();
    }));
  }

  render() {
    let articlesLoaded = this.data();
    let loaded = false;
    this.masonryLoader();

    if (articlesLoaded!==undefined && articlesLoaded.length==12) {
      setTimeout(function(){
        imagesLoaded('.grid', function() {
          let msnry = new Masonry( '.grid',{
            itemSelector: '.grid-item',
            columnWidth: '.grid-sizer'
          });
          msnry.reloadItems();
        });
      }, 300);
      loaded = true;
    }

    return ( !loaded ? ( <LoaderData />) : (
      <div className="grid" id="grid">
        <div className="grid-sizer" />
        {articlesLoaded}
      </div>
      )
    );
  }
}

Articles.propTypes = {
  actions: PropTypes.object.isRequired,
  articles: PropTypes.object.isRequired,
  currentPage: PropTypes.number.isRequired,
};


function mapStateToProps(state) {
  return {
    articles: state.Dashboard
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(dashboardActions, dispatch)
  };
}


export default connect(mapStateToProps, mapDispatchToProps)(Articles);
