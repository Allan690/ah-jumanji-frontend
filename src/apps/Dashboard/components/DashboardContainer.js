import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";

import SecondaryNav from './SecondaryNav';
import Articles from './Articles';
import Pagination from "../../Pagination/components/Pagination";

class DashboardContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1
    };
  }

  onPageChanged = data => {
    const { currentPage } = data;
    this.setState({ currentPage });
    };

  render() {
    const { currentPage } = this.state;
    const { articleCount } = this.props;
    return (
      <React.Fragment>
        <SecondaryNav />
        <div className="container">
          <h4 className="article-header">
            <i className="fas fa-align-right" />
            <span id="category">
              &nbsp;
              All Articles.
            </span>
          </h4>
          <Articles currentPage={currentPage} />
          <Pagination
            totalRecords={24}
            pageLimit={12}
            pageNeighbours={0}
            onPageChanged={this.onPageChanged}
        />
        </div>
      </React.Fragment>
    );
  }
}

DashboardContainer.propTypes = {
  articleCount: PropTypes.number.isRequired
};

const mapStateToProps = ({ Dashboard }) => Dashboard;

export default connect(mapStateToProps)(DashboardContainer);
