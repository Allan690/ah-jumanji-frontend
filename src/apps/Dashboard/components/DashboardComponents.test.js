import React from 'react';

import Enzyme, { shallow, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Articles from './Articles';
import LoaderData from '../../common/components/LoaderData';
import DashboardContainer from './DashboardContainer';
import ArticleLikesButton from './ArticleLikes';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('components', () => {
  describe('Articles', () => {
    it('should render self and subcomponents', () => {
      const store = mockStore({});
      const props = {
        articles: {
          articles: ""
        }
      };
      const enzymeWrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Articles {...props} />
          </MemoryRouter>
        </Provider>,);
      expect(enzymeWrapper.find('Articles')).toBeDefined();
      expect(enzymeWrapper.find('div')).toHaveLength(331);
      expect(enzymeWrapper.contains(<LoaderData />)).toBe(true);
    });
});
});


describe('<Articles />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<Articles debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('<ArticleLikesButton />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<ArticleLikesButton debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('components', () => {
  describe('DashboardContainer', () => {
    it('should render self and subcomponents', () => {
      const store = mockStore({});
      const enzymeWrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <DashboardContainer currentPage={1} />
          </MemoryRouter>
        </Provider>,);
      expect(enzymeWrapper.find('DashboardContainer')).toBeDefined();
      expect(enzymeWrapper.find('div')).toHaveLength(333);
    });
});
});
