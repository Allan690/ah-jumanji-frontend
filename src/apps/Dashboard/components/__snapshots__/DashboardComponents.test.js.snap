// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleLikesButton /> should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<Articles /> should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
